OJP/OJP_Trips.xsd

```diff
@@ -1072,9 +1084,30 @@
 					<xs:documentation>If true, then the request may contain object references from another system. Default is FALSE.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
-			<xs:element name="RefineLegRef" type="xs:NMTOKEN" maxOccurs="unbounded">
+			<xs:element name="RefineLegRef" type="xs:NMTOKEN" maxOccurs="unbounded" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>Refers to the legs to be refined by the server. If none is given, then all legs are open for refinement (depending if the relevant system can refine them).</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:group ref="TripRefinementDataFilterGroup"/>
+			<xs:element name="KeepLegs" type="xs:boolean" minOccurs="0" default="true">
+				<xs:annotation>
+					<xs:documentation>Shall the legs be kept or can neighbours be recalculated in one go and changed.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="RefitLegs" type="xs:boolean" minOccurs="0" default="false">
+				<xs:annotation>
+					<xs:documentation>With refit, the leg is/legs are recalculated from the end point of the adjascent legs. If necessary new foot paths are added.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="CheckAvailability" type="xs:boolean" minOccurs="0" default="false">
+				<xs:annotation>
+					<xs:documentation>Checks the availability of relevant legs depending on the mode/submode. Operators or system references must be set to do this (uses OJPAvailabilityRequest indirectly)</xs:documentation>
```

Comment:
Let us talk about the workflow you are expecting behind this on Friday.

Reply:
to be discussed label already added